### DISCUSSION TITLE
Rounded Image Corners

### DISCUSSION THREAD

**Discussion by mbilal234** [typst/typst#1977] [Q&A] [answered]
Question:
I am trying to implement a Carousel component in typst that is based on "Carousel" in Material3. 
For this carousel, it requires that the image has rounded corners.
Currently, I cannot seem to figure out a way to round the image corners like how we do with rect() using radius.

Is there any way to make an image with a background (.jpeg for example) have rounded corners to match the rounded corners of the box?
Any and all help will be appreciated.

Code:

#let Carousel(
    ..images
    ) = {
    let _images = images.pos()
    for img in _images{
        stack(dir:ltr,
            box(fill:colors.surface_container,
                img,
                radius:10pt
                )
            )
    }}

Reference:
A material3 reference is attached as to how the Carousel.typ should look like
![CarouselTest](https://github.com/typst/typst/assets/111046732/5ee0512d-b876-4b38-b306-351073e6adbb)

**Comment by frozolotl** [comment]
I don't think this is currently possible.*
I would like clipping to work in this case, but it does not, as of right now.

```typ
block(
  clip: true,
  radius: 10pt,
  image(path)
)
```

*I could imagine that a function that generates an SVG which embeds the image could work. It doesn't sound easy to do that though.

**Reply by Enivex** [reply]
You could emulate it by drawing the border on top of the image, matching the background color

**Comment by TieWay59** [comment]
![image](https://github.com/typst/typst/assets/40034603/e957e5b5-36e4-4f1b-878b-ffb134bd1abc)

@Enivex Am I doing right? it's a bit weird.

```typst
#box( 
  width: 65pt,
  height: 65pt,
  stroke: 1pt + black,
  inset: 5pt, 
)[
  #stack(
    spacing: -55pt,
    image(
      "avt.jpeg", 
      height: 55pt, 
      width: 55pt
    ),
    box(
      height: 55pt,
      width: 55pt,
      radius: 8pt,
      stroke: 8pt + white, 
    ),
  )
]

```

**Comment by Enivex** [comment]
@TieWay59 if you want fewer magic constants, you can do something like this
```
#let my-image = image("30c.jpg", width: 5cm)
#block[
  #my-image
  #style(styles => {
    place(top+left,box(stroke: white+0.5cm,radius:0.5cm,..measure(my-image,styles)))
  })
]
```
but it's not great regardless.

![image](https://github.com/typst/typst/assets/2191754/231c7a3a-8078-4216-a308-db7aace18296)

**Comment by interdisziplina** [comment, marked as answer]
For me, `clip: true` works on boxes now, so the first solution seems to be the best option for typst v0.11.0+.

(Just noting this in case anyone finds this from searching, as I did)

**Reply by rwmpelstilzchen** [reply]
Thanks! 🙂

@mbilal234, would you like to mark this as the answer, for people who find this question by searching Typst’s GitHub discussions or a general search engine?
